### DISCUSSION TITLE
Lualine section configuration

### DISCUSSION THREAD

**Discussion by siph** [nix-community/nixvim#369] [Q&A] [answered]
I am trying to get lualine to display relative paths.

The lua should look like this:
```lua
{'filename', path = 1}
```
But I can't figure out what my string should look like to produce this lua.

I've tried many variations along the lines of:
```nix
{
  plugins.lualine.sections.lualine_c = [ "filename" "path = 1" ];
}
```
and
```nix
{
  plugins.lualine.sections.lualine_c = [ "filename, path = 1" ];
}
```

**Comment by GaetanLepage** [comment]
Very important remark indeed !
So, there is a way to achieve those "hybrid dict-list" lua tables by doing the following:
```nix
{
  plugins.lualine.sections.lualine_c = {
    "@" = "filename";
    path = 1;
  };
}
```
The "`@`" is a special character in nixvim's objects: All the attrs keys that start with an `@` will be translated to a simple list item being the corresponding value.
Relevant code: https://github.com/pta2002/nixvim/blob/3600698abad00756947eb3a3d837beedf1c2f551/lib/helpers.nix#L22-L23

**Comment by siph** [comment]
I see, thanks for the reply. I did see this syntax in the source but didn't know what it meant. Unfortunately this solution results in an error because it isn't in a list (is this conversion supposed to happen automatically in the helper functions?):

```shell
error: A definition for option `plugins.lualine.sections.lualine_c' is not of type `null or (list of (string or (submodule)))'. Definition values:
       - In `/nix/store/5qjm9v9ifrwhzv7zywb71g04gmn7hl90-source/config/lualine.nix':
           {
             "@" = "filename";
             path = 1;
           }

```

Sticking it in a list also causes an error:

```shell
error: The option `plugins.lualine.sections.lualine_c."[definition 1-entry 1]"."@"' does not exist. Definition values:
       - In `/nix/store/jkbgyjjfzbqvkfzv325y8wbp4nqdzwvy-source/config/lualine.nix': "filename"
```

Do I need to define this option as a full module? Is it possible to accomplish this with a list of strings?

**Reply by traxys** [reply]
I believe this is a limitation of nixvim, we should allow an attrset in this position for this use case

**Comment by traxys** [comment, marked as answer]
I think you can do:
```nix
{
  plugins.lualine.sections.lualine_c = [{
    name = "filename";
    extraConfig = {
        path = 1;
    };
  }];
}
```

**Reply by pta2002** [reply]
This is the best way to do this. We have explicit support for lualine specifically because of this!
